addons/account/models/account_move.py

```diff
@@ -1462,7 +1462,7 @@ def _inverse_amount_total(self):
             to_write = []
 
             amount_currency = abs(move.amount_total)
-            balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.date)
+            balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.invoice_date or move.date)
```

Comment:
I assumed that the current tests would be sufficient because they already have different dates and invoice_dates with varying exchange rates but I can add more